waggle-bk-reverse-tunnel.sh

```diff
@@ -1,45 +1,66 @@
 #!/bin/bash -e
 
+# getconf returns the first value for a key in a config file and section
+# TODO(sean) factor this out so this code and tests can share it
+getconf() {
+    file="$1"
+    sec="$2"
+    key="$3"
+
+    awk -v matchsec="$sec" -v matchkey="$key" '
+# skip comments
+/^#/ { next }
+
+# track current section
+/^\[.*\]/ {
+    # removes outer [ ]
+    gsub(/\[|\]/, "", $1);
+    sec = $1;
+}
+
+# print everything to after = when we match section and key 
+($2 == "=") && (sec == matchsec) && ($1 == matchkey) {
+    for (i = 3; i <= NF; i++) {
+        printf $i
+        if (i < NF) {
+            printf " "
+        }
+    }
+    exit
+}
+' "$file"
+}
 
 VERSION="{{VERSION}}"  # do not edit
 
-
 CONFIG_FILE=/etc/waggle/config.ini
 
-
 id=$(</etc/waggle/node-id)
 
-
 if [ "${id}_" == "_" ] ; then
     echo "Error: Node id missing"
     exit 1
 fi
 
-
 if [ ! -e ${CONFIG_FILE} ] ; then
     echo "Error: Config file ${CONFIG_FILE} missing"
     exit 1
 fi
 
-CONFIG_SECTION=$(grep '^\[reverse-tunnel\]' -A 999 ${CONFIG_FILE} | tail -n +2  | grep -m1 -B 999 '^\[' | head -n -1)
+BK_HOST=$(getconf ${CONFIG_FILE} reverse-tunnel host)
+BK_PORT=$(getconf ${CONFIG_FILE} reverse-tunnel port)
+KEY_FILE=$(getconf ${CONFIG_FILE} reverse-tunnel key)
+SSH_OPTIONS=$(getconf ${CONFIG_FILE} reverse-tunnel ssh-options)
+SSH_OPTIONS="${SSH_OPTIONS:-vv}"
 
-match=host
-BK_HOST=$(echo "${CONFIG_SECTION}" | grep -m1 ^[[:space:]]*${match}[[:space:]]*= | cut -d '=' -f 2 |  tr -d ' ' )
 echo "BK_HOST=${BK_HOST}"
-
-match=port
-BK_PORT=$(echo "${CONFIG_SECTION}" | grep -m1 ^[[:space:]]*${match}[[:space:]]*= | cut -d '=' -f 2 |  tr -d ' ' )
 echo "BK_PORT=${BK_PORT}"
-
-match=key
-KEY_FILE=$(echo "${CONFIG_SECTION}" | grep -m1 ^[[:space:]]*${match}[[:space:]]*= | cut -d '=' -f 2 |  tr -d ' ' )
 echo "KEY_FILE=${KEY_FILE}"
-
-# source: https://stackoverflow.com/a/64993893/2069181
+echo "SSH_OPTIONS=${SSH_OPTIONS}"
 
 set -x
 
-ssh -vv \
+ssh ${SSH_OPTIONS} \
     -o "ServerAliveInterval 60" \
```

Comment:
can we put these options in the config too?

```
-o "ServerAliveInterval 60"
-o "ServerAliveCountMax 3"
```

Reply:
I would not see us these changing these options anytime soon (and they seem important), thus I think having them hardcoded is a bit safer. But config is also ok.

Reply:
I was thinking if we wanted to adjust the timing from 60 seconds to 5 minutes or from the max from 3 to 5 that it would be nice to have that be a config.  we could have it different per node based on environment or just if we want to make a change.

Reply:
If I do add those to the config, I'm thinking of just adding `keepalive-interval` and `keepalive-count` flags. I'm hesitant to put the whole `-o ...` thing into options since it starts to keep make the code "anemic" and harder to understand with having to also understand details of the current config.

Reply:
Your choice.  whatever you think works best.

Reply:
Of course, these will default to 60 and 3.